src/main/java/sg/ncl/DataController.java

```diff
@@ -203,6 +204,124 @@ public String removeDataset(@PathVariable String id, @PathVariable Optional<Stri
+                } else {
+                    log.warn("Unknown error for getting dataset access request.");
+                }
+            } else {
+                log.info("Dataset access request retrieved: {}", dataResponseBody);
+                DataAccessRequest dataAccessRequest = extractDataAccessRequest(dataResponseBody);
+                model.addAttribute("dataAccessRequest", dataAccessRequest);
+            }
+        } catch (IOException e) {
+            log.error("requestDataset: {}", e.toString());
+            throw new WebServiceRuntimeException(e.getMessage());
+        }
+
+        return "data_request_access";
+    }
+
+    @RequestMapping("{did}/requests/{rid}/approve")
+    public String approveRequest(@PathVariable String did,
+                                 @PathVariable String rid,
+                                 RedirectAttributes redirectAttributes) throws WebServiceRuntimeException {
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.getRequest(did, rid), HttpMethod.PUT, request, String.class);
+        String dataResponseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(dataResponseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+                if (exceptionState == FORBIDDEN_EXCEPTION) {
+                    log.warn("Approving of dataset access request forbidden.");
+                    redirectAttributes.addFlashAttribute(MESSAGE_ATTRIBUTE, error.getMessage());
+                } else if (exceptionState == DATA_ACCESS_REQUEST_NOT_FOUND_EXCEPTION) {
+                    log.warn("Dataset access request not found.");
+                    redirectAttributes.addFlashAttribute(MESSAGE_ATTRIBUTE, error.getMessage());
+                } else if (exceptionState == DATA_NOT_MATCH_EXCEPTION) {
+                    log.warn("Dataset access request does not have matching parent.");
+                    redirectAttributes.addFlashAttribute(MESSAGE_ATTRIBUTE, error.getMessage());
+                } else if (exceptionState == DATA_NOT_FOUND_EXCEPTION) {
+                    log.warn("Dataset not found for approving request.");
+                    redirectAttributes.addFlashAttribute(MESSAGE_ATTRIBUTE, error.getMessage());
+                } else {
+                    log.warn("Unknown error for approving dataset access request.");
+                }
+            } else {
+                log.info("Dataset access request approved: {}", dataResponseBody);
+                redirectAttributes.addFlashAttribute(MESSAGE_ATTRIBUTE, "Request Approved");
+            }
+        } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)